sql-plugin/src/main/scala/org/apache/spark/sql/hive/rapids/HiveProviderImpl.scala

```diff
@@ -226,6 +243,10 @@ class HiveProviderImpl extends HiveProvider {
                 dt => dt.isInstanceOf[TimestampType])
             }
 
+            // Check if datatypes are all supported.
+            tableRelation.dataCols.filter(hasUnsupportedType)
```

Comment:
nit, I think we can remove the `filter` and instead invoke the `hasUnsupportedType` function within `flagUnsupportedType`, and conditionally invoke `willNotWorkOnGpu`

Reply:
Gotcha. 
I thought having them separate would be more descriptive. I've changed it now.